main.nf

```diff
@@ -70,10 +72,39 @@ workflow {
       input_h5
     )
 
+    // Take the resulting files, split & group them by name
+    // Use the size and remainder arguments in groupTuple()
+    // to control the size of the inputs to the concat() process
+    profiler_results_ch = CellProfiler.out
+        .flatten()
+        .map({ i -> [ i.name, i ]})
+        .groupTuple(size: params.reduce_n , remainder: true)
+
+    // For each group of files, concatenate them together
+    ConcatFiles_Round1(
+        profiler_results_ch
+      )
+
+    // Take the results from the first round of concatenation,
+    // group them by name, so they can all be concatenated together
+    concat_ch = ConcatFiles_Round1.out
+        .flatten()
+        .map({ i -> [ i.name, i ]})
+        .groupTuple()
+
+    // Concatenate all files of the same name together
+    ConcatFiles_Round2(
+        concat_ch
+      )
+
 }
 
 process CellProfiler {
   container "cellprofiler/cellprofiler:${params.version}"
+  label 'mem_veryhigh'
+  maxForks 100
```

Comment:
Let's take this configuration out of the process definition and move it into the `nextflow.config` for the entire workflow. I'd like this to be our standard approach for workflow configurations, just keeping the `label` in the process definition.